micro-rdk/src/common/ota.rs

```diff
@@ -189,20 +192,38 @@ impl OtaService {
             );
             uri = hyper::Uri::from_parts(parts).map_err(|e| OtaError::Other(e.to_string()))?;
         };
-        let io = self
-            .connector
-            .connect_to(&uri)
-            .map_err(|e| OtaError::Other(e.to_string()))?
-            .await
-            .map_err(|e| OtaError::Other(e.to_string()))?;
 
-        let (mut sender, conn) = {
-            http2::Builder::new(self.exec.clone())
-                .max_frame_size(16_384) // lowest configurable value
-                .timer(H2Timer)
-                .handshake(io)
-                .await
-                .map_err(|e| OtaError::Other(e.to_string()))?
+        let (mut sender, conn) = loop {
+            match self.connector.connect_to(&uri) {
+                Ok(acceptor) => {
+                    // failed to initialize the acceptor for some reason
+                    match acceptor.await {
+                        Ok(io) => {
+                            match http2::Builder::new(self.exec.clone())
+                                .max_frame_size(16_384) // lowest configurable
+                                .timer(H2Timer)
+                                .handshake(io)
+                                .await
+                            {
+                                Ok(pair) => break pair,
+                                Err(e) => {
+                                    log::error!("failed to build request: {}", e);
+                                }
+                            }
+                        }
+                        Err(e) => {
+                            log::error!("failed to create tcp stream: {}", e);
+                        }
+                    }
+                }
+                Err(e) => {
+                    // log and continue
+                    log::error!("failed to initiate connection: {}", e);
+                }
```

Comment:
- Is there some way to structure this that isn't so branchy / procedural? Could the logs be generated via `inspect_err`?
- Like in the other review, let's take the time to make these log messages better.